dm/master/server.go

```diff
@@ -2191,3 +2171,47 @@ func (s *Server) HandleError(ctx context.Context, req *pb.HandleErrorRequest) (*
 		Sources: workerResps,
 	}, nil
 }
+
+// sharedLogic does some shared logic for each RPC implementation
+// arguments with `Pointer` suffix should be pointer to that variable its name indicated
+// return `true` means caller should return with variable that `xxPointer` modified
+func (s *Server) sharedLogic(ctx context.Context, req interface{}, respPointer interface{}, errPointer *error) bool {
+	pc, _, _, _ := runtime.Caller(1)
+	fullMethodName := runtime.FuncForPC(pc).Name()
+	methodName := fullMethodName[strings.LastIndexByte(fullMethodName, '.')+1:]
+
+	// origin code:
+	//	log.L().Info("", zap.Stringer("payload", req), zap.String("request", "ListMember"))
+	log.L().Info("", zap.Any("payload", req), zap.String("request", methodName))
+
+	// origin code:
+	//  isLeader, needForward := s.isLeaderAndNeedForward()
+	//	if !isLeader {
+	//		if needForward {
+	//			return s.leaderClient.ListMember(ctx, req)
+	//		}
+	//		return nil, terror.ErrMasterRequestIsNotForwardToLeader
+	//	}
+	isLeader, needForward := s.isLeaderAndNeedForward()
+	if !isLeader {
+		if needForward {
+			params := []reflect.Value{reflect.ValueOf(ctx), reflect.ValueOf(req)}
```

Comment:
How about adding a log to indicate we are forwarding the request from XXX to YYY?